# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 3b1dbe909aaf
# Full Hash: 3b1dbe909aaf1982c72db1e5bc8cb35208ae6c77
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-23 21:43:09
# Regressor Bug: 1616775
# File Overlap Count: 1
# Description:
#   Bug 1616775 - P4. Add CompleteAllowAccessFor IPDL r=timhuang,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71013
# ==============================================================================

diff -r 123b6c3a0ede -r 3b1dbe909aaf dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Thu Apr 23 14:28:41 2020 +0000
+++ b/dom/ipc/PContent.ipdl	Thu Apr 23 14:25:16 2020 +0000
@@ -123,6 +123,8 @@
 using refcounted class nsDocShellLoadState from "nsDocShellLoadState.h";
 using mozilla::dom::ServiceWorkerShutdownState::Progress from "mozilla/dom/ServiceWorkerShutdownState.h";
 using refcounted class mozilla::dom::CrossProcessSHEntry from "mozilla/dom/MaybeNewPSHEntry.h";
+using mozilla::ContentBlockingNotifier::StorageAccessGrantedReason from "mozilla/ContentBlockingNotifier.h";
+using mozilla::ContentBlocking::StorageAccessPromptChoices from "mozilla/ContentBlocking.h";
 
 union ChromeRegistryItem
 {
@@ -1466,6 +1468,14 @@
                                                   int aAllowMode)
           returns (bool unused);
 
+    async CompleteAllowAccessFor(MaybeDiscardedBrowsingContext aParentContext,
+                                 uint64_t aTopLevelWindowId,
+                                 Principal aTrackingPrincipal,
+                                 nsCString aTrackingOrigin,
+                                 uint32_t aCookieBehavior,
+                                 StorageAccessGrantedReason aReason)
+          returns (StorageAccessPromptChoices? choice);
+
     async StoreUserInteractionAsPermission(Principal aPrincipal);
 
    /**